connector/controllers/main.py

```diff
@@ -112,11 +112,10 @@ def retry_postpone(job, message, seconds=None):
             retry_postpone(job, unicode(err), seconds=err.seconds)
             _logger.debug('%s postponed', job)
 
-        except (FailedJobError, Exception):
+        except:
```

Comment:
AFAICS, `InvalidDataError` is a `ConnectorException, which is `RuntimeError` which is an `Exception`, so changing this exception handler should not have any influence on your use case. Your root cause must be something else.